### DISCUSSION TITLE
upload files w/ faraday 2

### DISCUSSION THREAD

**Discussion by dorkalev** [lostisland/faraday#1377] [Help]
well, no spec, no docs. how do we upload files with faraday 2?

**Comment by iMacTia** [comment]
Hi @dorkalev, you can use the [faraday-multipart](https://github.com/lostisland/faraday-multipart) middleware to upload files in Faraday v2.
It was previously bundled with Faraday, but it moved out with the latest major release.

If you're upgrading from Faraday v1, I'd also suggest to check out the [UPGRADING guide](https://github.com/lostisland/faraday/blob/main/UPGRADING.md) and the new [Awesome Faraday](https://github.com/lostisland/awesome-faraday/#adapters) repo to discover all sorts of useful middleware and adapters 🙌
